### DISCUSSION TITLE
Remember the passphrase for 'pgp'

### DISCUSSION THREAD

**Discussion by fastovenko** [tejado/obsidian-gpgCrypt#7] [Q&A] [answered]
First of all, thank you for the app. It's useful.
I wanted to use gpg to encrypt notes, but it doesn't work. I click "Encrypt with a key pair" in the context menu of the note, then close the note. When more than 5 seconds have passed, I open the note and see the entire text.

Please help me. What I need to do?

<img width="1185" alt="image" src="https://github.com/tejado/obsidian-gpgCrypt/assets/63233652/a4ed69b6-07f3-4aa1-8393-21a9e84eb619">

**Comment by tejado** [comment, marked as answer]
Hi, what do you mean with "I open the note and see the entire text"? On the Obsidian status bar (bottom right) should be a lock icon which indicates when a note is encrypted. You can also verify with another text editor if the file is encrypted.

EDIT: the remember passphrase setting is currently only effective for the OpenPGP.js backend. For GnuPG CLI Wrapper, gpg is getting executed so the passphrase will be typed into and remembered by the gpg agent. Sorry, this might be indeed confusing and I might need to change this.

**Reply by fastovenko** [reply]
I mean I can read all text in the note.
Yes, I see the lock icon on the bottom. I created a gpg key with a password. When I open the note again, the password is not requested. But I want to use one.

_When I used 'OpenPGP.js' a password was needed._

**Reply by tejado** [reply]
Then you have to configure the password cache in gpg itself, e.g. https://stackoverflow.com/questions/71164346/gpg-key-works-only-for-a-few-minutes (adjust the values to your needs)

**Reply by fastovenko** [reply]
I created the file '~/.gnupg/gpg-agent.conf':

```
default-cache-ttl 0
max-cache-ttl 0
```

I reloaded Obsidian, and the error appears in the right top corner when I try to open an encrypted note.

<img width="321" alt="image" src="https://github.com/tejado/obsidian-gpgCrypt/assets/63233652/e70f65b0-e3e9-4162-b1f3-8dbcaa459a25">

**Reply by tejado** [reply]
This is the output of gpg. Can you check maybe this: https://stackoverflow.com/questions/51504367/gpg-agent-forwarding-inappropriate-ioctl-for-device ?

**Comment by fastovenko** [comment]
I've decided to use 'OpenPGP.js' instead of 'pgp'. It's more convenient, and I won't need to set up or configure 'pgp' if I decide to move to another host. But I'm curious why you added support for 'pgp'? In what kind of cases would it be properly used?

**Reply by tejado** [reply]
If you already using gpg, it would be more convenient to use this as well for Obsidian regarding key management. Also it supports storing the keys on a smartcard.

**Reply by rubaboo** [reply]
I am using it with "GnuPG CLI Wrapper" backend, and it's awesome. Easily the best encryption plugin available for Obsidian. Only 6.4k downloads—it should be way more popular than that!

Perhaps that blurb under "Encryption backend" needs to be expanded, to say "intended **for advanced users**, offering more configuration options via **gpg agent configuration**, including **passphrase caching** and support for smartcards."
